grpc/src/builder.rs

```diff
@@ -131,12 +136,23 @@ impl GrpcClientBuilder {
     }
 
     /// Build [`GrpcClient`]
+    ///
+    /// Returns error if no transports were configured.
     pub fn build(self) -> Result<GrpcClient, GrpcClientBuilderError> {
-        let transport = match self.transport {
-            TransportSetup::EndpointUrl(url) => build_transport(url)?,
-            TransportSetup::BoxedTransport(transport) => transport,
-            TransportSetup::Unset => return Err(GrpcClientBuilderError::TransportNotSet),
-        };
+        if self.transports.is_empty() {
+            return Err(GrpcClientBuilderError::TransportNotSet);
+        }
+
+        let transports: Vec<BoxedTransport> = self
+            .transports
+            .into_iter()
+            .map(|entry| match entry {
+                TransportEntry::EndpointUrl(url) => build_transport(url),
+                TransportEntry::BoxedTransport(t) => Ok(t),
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let transports: std::sync::Arc<[BoxedTransport]> = transports.into();
```

Comment:
nit, I'd move this call to  `GrpcClient::new(transports.into(), ...)`, this way we don't need to specify the type